ts/redux/async_dispatcher.ts

```diff
@@ -58,15 +59,47 @@ export const asyncDispatcher = {
             const provider = web3Wrapper.getProvider();
             const contractAddresses = getContractAddressesForNetworkOrThrow(networkId as number);
             const zrxTokenContract = new ERC20TokenContract(contractAddresses.zrxToken, provider);
-            const [ethBalanceInWei, zrxBalance] = await Promise.all([
+            const [ethBalanceInWei, zrxBalance, zrxAllowance] = await Promise.all([
                 web3Wrapper.getBalanceInWeiAsync(address),
                 zrxTokenContract.balanceOf.callAsync(address),
+                zrxTokenContract.allowance.callAsync(address, contractAddresses.erc20Proxy),
             ]);
+
             dispatcher.updateAccountEthBalance({ address, ethBalanceInWei });
             dispatcher.updateAccountZrxBalance(zrxBalance);
+            dispatcher.updateAccountZrxAllowance(zrxAllowance);
         } catch (e) {
             logUtils.warn(e);
             return;
         }
     },
+
+    increaseZrxAllowanceAndDispatchToStoreIfNeededAsync: async (
+        providerState: ProviderState,
+        networkId: Network,
+        amountToStakeBaseUnits: BigNumber,
+        dispatcher: Dispatcher,
+    ) => {
+        const { provider } = providerState;
+        const ownerAddress = (providerState.account as AccountReady).address;
+        const gasInfo = await backendClient.getGasInfoAsync();
+        const gasPriceInGwei = new BigNumber(gasInfo.fast / 10);
+        const gasPriceInWei = gasPriceInGwei.multipliedBy(1000000000);
+
+        const contractAddresses = getContractAddressesForNetworkOrThrow(networkId as number);
+        const erc20ProxyAddress = contractAddresses.erc20Proxy;
+        const zrxTokenContract = new ERC20TokenContract(contractAddresses.zrxToken, provider);
+
+        const currentAllowance = await zrxTokenContract.allowance.callAsync(ownerAddress, erc20ProxyAddress);
+
+        // TODO: some information modal needed?
+        if (currentAllowance.isLessThan(amountToStakeBaseUnits)) {
+            // tslint:disable:await-promise
+            await zrxTokenContract.approve.awaitTransactionSuccessAsync(erc20ProxyAddress, amountToStakeBaseUnits, {
+                from: ownerAddress,
+                gasPrice: gasPriceInWei,
+            });
```

Comment:
We usually do unlimited allowance, so it is more of a binary thing. 
https://github.com/0xProject/0x-starter-project/blob/f426e1c60e5bceb8848ca7fb618075289df9be09/src/scenarios/fill_order_sra.ts#L59